src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -217,6 +213,21 @@ Definitions.CHECKSTYLE_BUNDLE, getClass(),
         return instance;
     }
 
+    /**
+     * Calls method that does classpath scanning if required.
+     * This operation is very slow so it should be done only once.
+     *
+     * @return the map of third party Checkstyle module names to the set of their fully qualified
+     *      names
+     */
+    private Map<String, Set<String>> getThirdPartyModulesWithInitializationIfRequired() {
+        if (thirdPartyNameToFullModuleNames == null) {
+            thirdPartyNameToFullModuleNames =
+                    generateThirdPartyNameToFullModuleName(moduleClassLoader);
+        }
+        return thirdPartyNameToFullModuleNames;
+    }
```

Comment:
This implementation is better if we have no need for lazy loading elsewhere.

Reply:
I am ok with this. This is pretty easy to read. Lambdas aren't helping in any way here. Simple, straightforward code, no need of lambdas.

Reply:
@rnveach , please review this, if you approve it, we will merge this PR.

Reply:
I already merged the other PR which was opened. It was not clear I was to pick between these.

Reply:
@rnveach , just share your opinion on this update, if you like it more I can rebase it  to keep this code

Reply:
I prefer us to write a test to cover lazy loading than either method.

Reply:
It is next step with fancy mocking.
If we start doing it, this code will be easier to mock i think.

We can even test it by Junit with call of private method.

Reply:
Since the other PR is merged, I don't see a point in doing any steps except reverting the code and switching to an actual test to cover lazy loading, or an explanation why we can't create a test and keep the code as it is now.